controllers/chat/ChatController.js

```diff
@@ -426,6 +426,135 @@ module.exports = {
       return ErrorResponse.e400(res, error.message);
     }
   },
+  initChatMoveToAnon: async (req, res) => {
+    const {
+      targetUser,
+      message,
+      anon_user_info_color_code,
+      anon_user_info_color_name,
+      anon_user_info_emoji_code,
+      anon_user_info_emoji_name
+    } = req.body;
+
+    let members = [targetUser];
+    if (!members.includes(req.userId)) members.push(req.userId);
+
+    const client = StreamChat.getInstance(process.env.API_KEY, process.env.SECRET);
+    try {
+      /**
+       * @type {import('stream-chat').OwnUserResponse}
+       */
+      const user = {
+        name: `Anonymous ${anon_user_info_emoji_name}`,
+        id: req.userId,
+        image: '',
+        username: `Anonymous ${anon_user_info_emoji_name}`
+      };
+      await client.connectUser(user, req.token);
+
+      if (client.user.name !== `Anonymous ${anon_user_info_emoji_name}`) {
+        await client.upsertUser({id: req.userId, name: `Anonymous ${anon_user_info_emoji_name}`});
+      }
+
+      const channel = client.channel('messaging', {members});
+      const createdChannel = await channel.create();
+
+      await channel.updatePartial({
+        set: {
+          channel_type: CHANNEL_TYPE.ANONYMOUS,
+          anon_user_info_color_code,
+          anon_user_info_color_name,
+          anon_user_info_emoji_code,
+          anon_user_info_emoji_name
+        }
+      });
```

Comment:
This will throw error if current user role is not owner and not moderator. Please add try catch to this.